rocky/reports/report_types/name_server_report/report.py

```diff
@@ -0,0 +1,101 @@
+from dataclasses import dataclass
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict, List
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import IPAddressV4, IPAddressV6
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class NameServerCheck:
+    no_uncommon_ports: bool = False
+    has_dnssec: bool = False
+    has_valid_dnssec: bool = False
+
+
+@dataclass
+class NameServerChecks:
+    checks: List[NameServerCheck]
+
+    @property
+    def no_uncommon_ports(self):
+        return sum([check.no_uncommon_ports for check in self.checks])
+
+    @property
+    def has_dnssec(self):
+        return sum([check.has_dnssec for check in self.checks])
+
+    @property
+    def has_valid_dnssec(self):
+        return sum([check.has_valid_dnssec for check in self.checks])
+
+    def __len__(self):
+        return len(self.checks)
```

Comment:
I don't see any added value for this one (except for the rest as a general wrapper with helper utils) as it seems unused, but in what context will this be used?

Reply:
It's used in the templates